connectors/src/monitoring/index.ts

```diff
@@ -0,0 +1,45 @@
+export class MonitoringError extends Error {}
+import StatsDClient from "hot-shots";
+
+import mainlogger from "@connectors/logger/logger";
+const logger = mainlogger.child({ module: "temporal-monitoring" });
+
+export const statsDClient = new StatsDClient({});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const withMonitoring = <T extends Array<any>, U>(
+  fn: (...args: T) => U
+) => {
+  return async (...args: T): Promise<U> => {
+    let result: U | undefined = undefined;
+
+    try {
+      const startTime = new Date();
+      logger.info({ activity: fn.name }, `Activity started`);
```

Comment:
I fear `fn.name` is a bit too much magic in particular because there is no guarantee it aligns well with the prefix to the workflows IDs we use which would be a desirable property?

But I'm OK with starting with that. wdyt @fontanierh ?

Main question is, is there possibly other value we would want to pass in? eg parent workflow name maybe even the workflow ID (I presume we have access to it when we call the activity? 

In which case we should probably pass both the workflowName and activityName and anything else as explicit string parameter?

Otherwise LGTM. If no action on this, feel free to merge :+1:

Reply:
> But I'm OK with starting with that. wdyt @fontanierh ?

I think it would be cool to allow passing a `Record<string, string>` to `withMonitoring` and we add everything in it to the logger context

Reply:
We can manage to pass extra arguments to the activity that has been wrapper but there is no way to pass them to withMonitoring() and have them serialized and passed to the worker. Or if it's possible that would be very complex / opaque / risky.